sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_models.py

```diff
@@ -1032,3 +1197,276 @@ def _from_generated(cls, score):
+
+    def to_generated(self):
+        return _v3_2_preview_1_models.SentimentTask(
+            enable=True,
+            parameters=_v3_2_preview_1_models.SentimentTaskParameters(
+                model_version=self.model_version,
+                string_index_type=self.string_index_type,
+                opinion_mining=self.enable_opinion_mining
+            )
+        )
+
+
+class SentimentAnalysisTaskResult(DictMixin):
+    """SentimentAnalysisTaskResult contains the results of a single Sentiment Analysis task, including additional 
+        task metadata.
+
+    :ivar str name: The name of the task.
+    :ivar results: The results of the analysis.
+    :vartype results: list[~azure.ai.textanalytics.AnalyzeSentimentResult]
+    """
+
+    def __init__(self, **kwargs):
+        self.name = kwargs.get("name", None)
+        self.results = kwargs.get("results", [])
+
+    def __repr__(self, **kwargs):
+        return "SentimentAnalysisTaskResult(name={}, results={})" \
+            .format(self.name, repr(self.results))[:1024]
+
+
+class TextAnalysisResult(DictMixin):
+    """TextAnalysisResult contains the results of multiple text analyses performed on a batch of documents.
+
+    :ivar entities_recognition_results: A list of objects containing results for all Entity Recognition tasks
+        included in the analysis.
+    :vartype entities_recognition_results: list[~azure.ai.textanalytics.EntitiesRecognitionTaskResult]
+    :ivar pii_entities_recognition_results: A list of objects containing results for all PII Entity Recognition
+        tasks included in the analysis.
+    :vartype pii_entities_recogition_results: list[~azure.ai.textanalytics.PiiEntitiesRecognitionTaskResult]
+    :ivar entity_linking_results: A list of objects containing results for all Entity Linking tasks included in 
+        the analysis.
+    :vartype entity_linking_results: list[~azure.ai.textanalytics.EntityLinkingTaskResult]
+    :ivar key_phrase_extraction_results: A list of objects containing results for all Key Phrase Extraction tasks 
+        included in the analysis.
+    :vartype key_phrase_extraction_results: list[~azure.ai.textanalytics.KeyPhraseExtractionTaskResult]
+    :ivar sentiment_analysis_results: A list of objects containing results for all Sentiment Analysis tasks 
+        included in the analysis.
+    :vartype sentiment_analysis_results: list[~azure.ai.textanalytics.SentimentAnalysisTaskResult]
+    """
+    def __init__(self, **kwargs):
+
+        self.entities_recognition_results = kwargs.get("entities_recognition_results", [])
+        self.pii_entities_recognition_results = kwargs.get("pii_entities_recognition_results", [])
+        self.entity_linking_results = kwargs.get("entity_linking_results", [])
+        self.key_phrase_extraction_results = kwargs.get("key_phrase_extraction_results", [])
+        self.sentiment_analysis_results = kwargs.get("sentiment_analysis_results", [])
```

Comment:
question out of curiosity: why is there no language detection option to the analyze endpoint? I think i vaguely remember hearing this, but can't remember the details

Reply:
I think we talked about this a few days ago, and I'm not completely sure about the reason but I think it probably has something to do with the use of MultiLanguageBatchInput for the current and planned supported services for Analyze.